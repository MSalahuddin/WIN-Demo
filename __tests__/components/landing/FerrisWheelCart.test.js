import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import FerrisWheelCart from '../../../js/components/landing/FerrisWheelCart';
import MockProvider from '../../MockProvider';

describe('Component FerrisWheelCart', () => {
  const action = jest.fn();
  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <FerrisWheelCart prizeImage={1} prizeTitle="prize" buttonAction={action} buttonTitle="button" />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when ferris wheel button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <FerrisWheelCart prizeImage={1} prizeTitle="prize" buttonAction={action} buttonTitle="button" />
      </MockProvider>
    );
    fireEvent(getByTestId('ferris-wheel-button'), 'press');
    expect(action).toHaveBeenCalled();
  });
});
