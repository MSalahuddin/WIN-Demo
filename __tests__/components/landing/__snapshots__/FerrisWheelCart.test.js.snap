// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FerrisWheelCart renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 518,
          "paddingTop": 100,
          "width": 426,
        },
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "paddingTop": 40,
            "zIndex": -10,
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 208,
              "width": 186,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": -10,
            "paddingHorizontal": 30,
          },
        ]
      }
    >
      <Text
        adjustsFontSizeToFit={true}
        alignCenter={false}
        allowFontScaling={true}
        color="#FFFFFF"
        ellipsizeMode="tail"
        fontFamily="SEMI BOLD"
        isUnderlined={false}
        isUppercase={false}
        maxFontSizeMultiplier={1.3}
        numberOfLines={1}
        size="XXSMALL"
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Calibre-Semibold",
              "fontSize": 21,
              "justifyContent": "center",
              "textAlign": "left",
              "textDecorationColor": "#FFFFFF",
              "textDecorationLine": "none",
              "textDecorationStyle": "solid",
              "textTransform": "none",
            },
          ]
        }
      >
        prize
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginTop": 0,
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "marginHorizontal": 10,
              "marginVertical": 10,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(96,202,142,1)",
              "borderRadius": 80,
              "height": 90,
              "justifyContent": "center",
              "opacity": 1,
              "width": 280,
            }
          }
          testID="ferris-wheel-button"
        >
          <View
            borderRadius={40}
            colorScheme="rgba(128,229,173,1)"
            height={45}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(128,229,173,1)",
                  "borderRadius": 80,
                  "height": 80,
                  "justifyContent": "center",
                  "width": 274,
                },
              ]
            }
            width={140}
          >
            <View
              borderRadius={40}
              colorScheme="rgba(114,213,158,1)"
              height={45}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(114,213,158,1)",
                    "borderRadius": 100,
                    "height": 76,
                    "justifyContent": "center",
                    "marginTop": 8,
                    "width": 274,
                  },
                ]
              }
              width={140}
            >
              <Text
                alignCenter={false}
                allowFontScaling={true}
                color="#FFFFFF"
                fontFamily="MEDIUM"
                isUnderlined={false}
                isUppercase={false}
                maxFontSizeMultiplier={1.3}
                size="SMALL"
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Calibre-Medium",
                      "fontSize": 27,
                      "justifyContent": "center",
                      "textAlign": "left",
                      "textDecorationColor": "#FFFFFF",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                      "textTransform": "none",
                    },
                    Object {
                      "marginTop": 10,
                    },
                  ]
                }
              >
                button
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
